include/links-permalinks.php

```diff
@@ -60,6 +60,47 @@ public function __construct( &$model ) {
 		$permalink_structure = get_option( 'permalink_structure' );
 		$this->root = preg_match( '#^/*' . $this->index . '#', $permalink_structure ) ? $this->index . '/' : '';
 		$this->use_trailing_slashes = ( '/' == substr( $permalink_structure, -1, 1 ) );
+
+		if ( did_action( 'pll_init' ) ) {
+			$this->init();
+		} else {
+			add_action( 'pll_init', array( $this, 'init' ) );
+		}
+	}
+
+	/**
+	 * Initializes permalinks hooks.
+	 *
+	 * @since 3.5
+	 *
+	 * @return void
+	 */
+	public function init() {
+		if ( did_action( 'wp_loaded' ) ) {
+			$this->can_prepare_rewrite_rules();
+		} else {
+			add_action( 'wp_loaded', array( $this, 'can_prepare_rewrite_rules' ), 9 ); // Just before WordPress callback `WP_Rewrite::flush_rules()`.
```

Comment:
This could be moved to the constructor.

Reply:
Done in b6eb780a.

Reply:
Found that's needed for our other plugins (especially for the test suites).